### PR TITLE
Add amount of disk space used by build to the log

### PR DESCRIPTION
**Pull request by sxa**
This will help log information that we can give to end users to tell how much space is required for builds, and also help with capacity planning on machines

### CODE REVIEW

**Comment by sxa** [comment]
run tests

**Review by karianna — APPROVED** [review]

**Comment by karianna** [comment]
Looks like TKG is broken - unrelated

**Review by steelhead31 — APPROVED** [review]
LGTM

**Comment by sxa** [comment]
run tests

**Comment by sxa** [comment]
FYI to reviewers the new push is to fix quoting which was a mess so hopefully doesn't affect you're review :-)

**Review by karianna — APPROVED** [review]

**Comment by karianna** [comment]
@sxa not sure why the checks are all failing..

**Comment by sxa** [comment]
See separate new issue referenced above for the GitHub check problems.

**Comment by sxa** [comment]
Switching this to draft just now becuase it's not working. When I run `make-adopt-build-farm` it gets to the end of `build.sh` and then stops after `build.sh : 09:56:13 : All done!` so it's not falling back to this script - maybe the result of an exec somewhere? Another commit will be required for this to work as expected (Or it could potentially be moved into `build.sh` however it's more for our infrastructure's benefit so this feels like the right place for it)

**Comment by sxa** [comment]
Rebasing to see if any of the actions failures fix themselves ...

**Comment by karianna** [comment]
only the gpg failure on centos7

File: build-farm/make-adopt-build-farm.sh
Comment (andrew-m-leonard) [inline build-farm/make-adopt-build-farm.sh:259]: Could we also dump the current volume's free disk space? This could help figure the github runner issue?

Comment (andrew-m-leonard) [reply build-farm/make-adopt-build-farm.sh:259]: Could we also print this info at the start of the build as well?

Comment (sxa) [reply build-farm/make-adopt-build-farm.sh:259]: I considered that but I'd rather not do that in this PR because the before and after data is potentially misleading for docker containers which use shared space, plus we have Nagios monitoring to detect problems in that area
2. For the github runner case, presumably you could do a subsequent PR with the change you want - that would allow you to test it in the GHA checks even if you don't end up merging it.

**Comment by sxa** [comment]
> only the gpg failure on centos7

There are still a few issues in this area which I believe others are covering at the moment. On the basis that this PR is (hopefully clearly) not the cause of them I'm happy to merge this with those checks failing.

**Comment by sxa** [comment]
> Switching this to draft just now becuase it's not working. When I run `make-adopt-build-farm` it gets to the end of `build.sh` and then stops after `build.sh : 09:56:13 : All done!` so it's not falling back to this script - maybe the result of an exec somewhere?

This was user error on my part as I hadn't set the `WORKSPACE` directory before running my local test. It is working as expected. IMHO this is good to merge (@andrew-m-leonard if you're ok to approve given [my earlier response to you](https://github.com/adoptium/temurin-build/pull/3569#discussion_r1482821362) that would be appreciated.

**Comment by sxa** [comment]
@andrew-m-leonard I'm going to merge this now since it has two approvers and your suggestions do not conflict with this. We can do a subsequent PR if you want to add the `df` somewhere.

File: build-farm/make-adopt-build-farm.sh
Comment (andrew-m-leonard) [reply build-farm/make-adopt-build-farm.sh:259]: yes makes sense

**Pull request merged**
